sql/src/main/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitor.java

```diff
@@ -233,8 +240,24 @@ public Projector visitOrderedTopN(OrderedTopNProjection projection, Context cont
         for (int i = numOutputs; i < inputs.size(); i++) {
             orderByIndices[idx++] = i;
         }
+
+        List<? extends Symbol> standaloneOutputs = projection.outputs().stream()
+            .filter(s -> s.symbolType() == SymbolType.INPUT_COLUMN ||
+                         s.symbolType() == SymbolType.LITERAL)
+            .collect(Collectors.toList());
+        List<DataType> standaloneOutputsDataTypes = Symbols.typeView(standaloneOutputs);
+
+        List<? extends Symbol> additionalSymbols = new ArrayList<>(projection.outputs());
+        additionalSymbols.removeAll(standaloneOutputs);
+        int rowMemoryOverhead = 32; // priority queues implementation are backed by an arrayList
+        // account for the functions in the outputs as overhead
+        for (Symbol additionalSymbol : additionalSymbols) {
+            rowMemoryOverhead += SizeEstimatorFactory.create(additionalSymbol.valueType()).estimateSize(null);
```

Comment:
not sure if it's worth caching the size estimators. @mfussenegger thoughts?

Reply:
Probably not, but I'm wondering if calculating a fixed overhead is good enough, as we won't get the "value" dependent size guesses for example for strings.

How about making `RowAccounting` generic, e.g `RowAccounting<T>` with a implementation that takes a `T -> int` function (`calculateSize`). 

Then we'd have a `Row -> int` and a `Object[] -> int` function, both utilizing the size estimators.